Release notes landing page

crhistianramirez opened the pull request.

djsteinmetz commented on src/pages/api-release-notes.tsx:
```diff
@@ -98,7 +113,7 @@ export default function ReleaseNotesListComponent(
   return (
     <Layout>
-      <Container maxWidth="md">
+      <Container maxWidth="lg">
```
Should we try and keep all the element wrappers consistent with the main 'DocBody' component, utilizing the `marginLeft: theme.spacing(9)`, and `marginRight: theme.spacing(DrawerWidthSpacingLg)`.  That way, the content is contained exactly, and it allows us to reduce the amount of nested elements.

crhistianramirez replied:
would that be instead of using this maxWidth option? Sorry im a newbbbb

djsteinmetz replied:
Well, it would just involve removing the `Container` component, as well as  both `Grid` components, and using a `Box` component to give the margin~s~ (we only need `marginLeft` here, since the `RightMenu` doesn't show up on this page).  Then you would give it a `maxWidth` on that Box if you didn't want it spanning the whole page.  But that way it's 1 wrapper, not 3.

crhistianramirez replied:
got it! i'll take a crack at that

crhistianramirez commented:
updated this pr.

I moved margin top and margin left from the child templates to the layout level so we only have one place to worry bout setting those.
